internal/proxy/task_search.go

```diff
@@ -506,10 +734,16 @@ func (t *searchTask) initSearchRequest(ctx context.Context) error {
 		}
 	}
 
-	if t.requery {
-		plan.OutputFieldIds = nil
+	vectorOutputFields := lo.Filter(t.schema.GetFields(), func(field *schemapb.FieldSchema, _ int) bool {
+		return lo.Contains(t.translatedOutputFields, field.GetName()) && typeutil.IsVectorType(field.GetDataType())
+	})
+	t.needRequery = len(vectorOutputFields) > 0
+	if t.needRequery {
+		plan.OutputFieldIds = t.functionScore.GetAllInputFieldIDs()
```

Comment:
Still that question: can we ensure all output fields user might specify can be covered by this function fields?

And also how it deal with the vector output. Since in original design vector need to be requery instead of direct searching.